packages/@aws-cdk/aws-cloud9-alpha/lib/environment.ts

```diff
@@ -257,6 +257,26 @@ export class Owner {
     return { ownerArn: user.userArn };
   }
 
+  /**
+   * Make an IAM assumed role the environment owner
+   *
+   * @param accountId The AccountId of the target account
+   * @param roleName The name of the assumed role
+   */
+  public static assumedRole(accountId: string, roleName: string): Owner {
```

Comment:
Why not user [IRole](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam.Role.html)?

```suggestion
  public static assumedRole(accountId: string, roleName: string): Owner {
```

Reply:
I did use `IRole` first (see [this commit](https://github.com/aws/aws-cdk/pull/27001/commits/7848cc36c246ed2a2806bcf4deea1fe6b2929f07)) but this creates ARNs that are not expected by Cloud9.

Cloud9 only allows the user to be of type `root`, `user`, `federated-user` and `assumed-role` (see [documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloud9-environmentec2.html#cfn-cloud9-environmentec2-ownerarn)).

The `IRole` interface is to broad for this and would allow developers to use any Role as the Cloud9 owner. This would not work due to above C9 constraints and ultimately lead to a failed development.

I would prefer to use a CDK construct here, but I don't think there is a CDK IAM construct that always resolves to `arn:*/assumed-role/*`.

Please let me know if there is one